### PR TITLE
[PDI-16671]-"Something Went Wrong" error message shows up when trying to Inspect Data which has a field name with Double Quotes.

### PR DESCRIPTION
**Pull request by denisprotopopov**
- additional fix. for pivot table

### CODE REVIEW

**Comment by denisprotopopov** [comment]
@pamval @mbatchelor please review the changes

**Comment by denisprotopopov** [comment]
@bmorrise  @pamval @mbatchelor please review the changes

**Comment by denisprotopopov** [comment]
@bmorrise @pamval @mbatchelor please review the changes

File: pdi-dataservice-client/src/main/java/org/pentaho/di/core/jdbc/ThinUtil.java [hunk 1]
```diff
@@ -655,4 +655,38 @@ public static String resolveFieldName( String field, RowMetaInterface serviceFie
     }
     return Optional.ofNullable( valueMeta );
   }
+
+  /**
+  * Remove first and last quote if string contains double quote
+  * */
+  public static String unQuote( String str ) {
+    StringBuilder builder = new StringBuilder( str );
+    if ( builder.indexOf( "\"\"" ) > 0 &&  builder.length() > 0 && builder.charAt( 0 ) == '"'
```
Comment (marcbatchelor-nucleus) [inline pdi-dataservice-client/src/main/java/org/pentaho/di/core/jdbc/ThinUtil.java]: Please replace this with:
    if ( str.startsWith("\"") && str.endsWith("\"")) {
      return str.substring(1, str.length() - 1 );
    } else {
      return str;
    }

Performance will be much better than using indexof and creating an intermediate object.

File: pdi-dataservice-client/src/main/java/org/pentaho/di/core/jdbc/ThinUtil.java [hunk 2]
```diff
@@ -655,4 +655,38 @@ public static String resolveFieldName( String field, RowMetaInterface serviceFie
...
+  }
+
+  /**
+   * Replaces each substring of this string that matches the given regular expression with the given replacement.
+   * @param strSrc source string
+   * @param strA expression to which this string is to be matched
+   * @param replacement the string to be substituted for each match
+   *
+   * */
+  public static String replaceSubString( String strSrc, String strA, String replacement ) {
```
Comment (marcbatchelor-nucleus) [inline pdi-dataservice-client/src/main/java/org/pentaho/di/core/jdbc/ThinUtil.java]: So why do this when String has a replaceAll method that accepts a regular expression ?
https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#replaceAll(java.lang.String,%20java.lang.String)

**Review by marcbatchelor-nucleus — CHANGES_REQUESTED** [review]

**Comment by denisprotopopov** [comment]
@mbatchelor please review the changes.

**Comment by denisprotopopov** [comment]
@mbatchelor please review the changes.

**Comment by denisprotopopov** [comment]
@bmorrise @pamval @mbatchelor please review the changes

**Comment by denisprotopopov** [comment]
@bmorrise @pamval @mbatchelor please review the changes

**Comment by denisprotopopov** [comment]
@bmorrise @pamval @mbatchelor please review the changes

**Pull request merged**